app/controllers/teachers_controller.rb

```diff
@@ -0,0 +1,9 @@
+class TeachersController < ApplicationController
+  def index
+    application = Application.find_by(user_id: current_user.id);
```

Comment:
You should have a `current_user` object in this controller. Since an application `belongs_to` a user, you can just say: `current_user.application`. You don't event need to set that to a variable.